### DISCUSSION TITLE
EdgeX device Provisioning using X.509 or Symmetric Key

### DISCUSSION THREAD

**Discussion by sirisatwika** [edgexfoundry/community#76] [Q&A]
We are currently working on using EdgeX Foundry for IOT Solution for that we are working on securely provisioning device through EdgeX using either X.509 or Symmetric Key, we are able to create and store Certificates using EdgeX Vault services. We are struck at how to add the device to EdgeX using the Certificate Created. Can you please help on how to achieve this if it is possible.

**Comment by bnevis-i** [comment]
This question is pretty vague and key provisioning is usually pretty device-specific.  I can't tell from the description whether you are trying to seed a key into a device, process a CSR, do some kind of mutual authentication, et cetera.  I would expect key provisioning to be done way before hooking it up to a production network and using it with EdgeX.
